x-pack/plugins/canvas/canvas_plugin_src/functions/common/as.ts

```diff
@@ -4,9 +4,17 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+// @ts-ignore untyped Elastic library
 import { getType } from '@kbn/interpreter/common';
+import { Datatable, ContextFunctionSpec } from '../types';
 
-export function asFn() {
+interface Arguments {
+  name: string;
+}
+
+type Context = string | boolean | number | null;
```

Comment:
We have several variations around `string | boolean | number`, maybe worth centralizing and giving them suitable names eg. `Primitive` etc. to elucidate on the meaning

Reply:
It would be confusing to do that, because then I'd have to write a string conversion for `Primitive`, so you could do `types: ['string', 'boolean', 'number']`... and there are only eight instances of this type at the moment anyway.

Reply:
Got it! I thought it'd be as simple as 
```
export type Primitive = string | boolean | number | null;
...
import { Primitive } from ...
type Context = Primitive
```
as it's an alias, but apparently that's not the case. Anyway indeed the repetition causes no issue whatsoever as these things won't change a lot